### DISCUSSION TITLE
dimension of RunUMAP in the Merging objects vignette

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#553] [Q&A] [answered]
Hello. I am currently working on merging two scATAC dataset following the Merging objects vignette (https://satijalab.org/signac/articles/merging.html) with a common peak set . However, I have a couple of questions in terms of "combined <- RunUMAP(combined, dims = 2:50, reduction = 'lsi')"

1. Every time I run this code, I get this warning message "Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'" . Does this matter? Do I have to edit it?
2. what is the meaning of dims = 2:50? Is it just like using the PCs in the PCA? 2:50 means from 2nd to 50th PCs?
3. if my understanding of Q1 is correct. why first dimension is not used? Because first dimension is highly correlated with sequencing depth?
4. unfortunately, when I use dims as 2:50, the dots of my two datasets which are actually two replicates in the  plot created by "DimPlot(combined, group.by = 'dataset', pt.size = 0.1)"  is not overlapping well. Do you have any comments or suggestions?
![Screen Shot 2021-04-08 at 6 26 49 AM](https://user-images.githubusercontent.com/54915282/114018925-6890b480-9833-11eb-92f5-40aa9bc44143.png)
But when I use dims as 1:5, they are overlapping better. But I am afraid it does not make sense?
![Screen Shot 2021-04-08 at 6 27 48 AM](https://user-images.githubusercontent.com/54915282/114019033-8bbb6400-9833-11eb-9105-aef5290970bd.png)

By the way, I am suffering from this error "#Error in `colnames<-`(`*tmp*`, value = start(x = region):end(x = region)) : 
#attempt to set 'colnames' on an object with less than two dimensions" When I run "CoveragePlot(
  object = combined,
  group.by ='dataset',
  region = "chr1-21552-274306278"
)". I do not know how to demonstrate it to you or make a reproducible example to you. I feel like my dataset are the same or extremely similar to the dataset used in the vignette and it is processed by cell ranger atac.
Can you give me some help or suggestions?

Thanks
Best

**Comment by timoast** [comment, marked as answer]
> Every time I run this code, I get this warning message "Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'" . Does this matter? Do I have to edit it?

I can't really give a better explanation than what the warning itself already says. It does not matter unless you want to perfectly reproduce results from an earlier version of Seurat.

> what is the meaning of dims = 2:50? Is it just like using the PCs in the PCA? 2:50 means from 2nd to 50th PCs?

Yes, see the documentation for `RunUMAP()`

> if my understanding of Q1 is correct. why first dimension is not used? Because first dimension is highly correlated with sequencing depth?

Yes, see the Signac vignettes for more information

> unfortunately, when I use dims as 2:50, the dots of my two datasets which are actually two replicates in the plot created by "DimPlot(combined, group.by = 'dataset', pt.size = 0.1)" is not overlapping well. Do you have any comments or suggestions? But when I use dims as 1:5, they are overlapping better. But I am afraid it does not make sense?

See "How should I decide on the number of dimensions to use?" in the [FAQ](https://satijalab.org/signac/articles/faq.html#how-should-i-decide-on-the-number-of-dimensions-to-use-)

> By the way, I am suffering from this error "#Error in colnames<-(*tmp*, value = start(x = region):end(x = region)) :

You should open a separate issue with the full code and the output of `sessionInfo()`.

**Reply by pengxin2019** [reply]
Hi Tim, 
Thanks for comments. Do you think it possible to find out the cells which are not overlapping well and  genetic regions which leads to the non-overlapping between two datasets in the plot above? Any suggestions?

**Reply by pengxin2019** [reply]
I mean in the plot when set up dims as 2:5( I noticed that 50 is the maximum i can choose. So, this plot should be the one explaining most of the variation and the best one. I can not increase dims anymore)
